src/state/apis/swapper/helpers/getInputOutputRatioFromQuote.ts

```diff
@@ -73,8 +73,9 @@ const _convertCryptoBaseUnitToUsdPrecision = (
   amountCryptoBaseUnit: BigNumber.Value,
 ): BigNumber => {
   const usdRate = selectUsdRateByAssetId(state, asset.assetId)
-  if (usdRate === undefined) throw Error(`missing usd rate for assetId ${asset.assetId}`)
-  return bnOrZero(fromBaseUnit(amountCryptoBaseUnit, asset.precision)).times(usdRate)
+  // TODO(gomes): revert me once we have a Portals market-data provider, this allows us to get quotes despite missing market data
+  // if (usdRate === undefined) throw Error(`missing usd rate for assetId ${asset.assetId}`)
```

Comment:
question: So we don't have any solution to bypass this only for portals? I'm not so happy having this security removed for every usages of `_convertCryptoBaseUnitToUsdPrecision` because of one swapper

Reply:
We absolutely could, also with @0xApotheosis revamp of custom/assets market data, we should now handle this gracefully in the UI so in favor of keeping this as-is